app/models/achievement.rb

```diff
@@ -52,21 +44,8 @@ class Achievement < ApplicationRecord
       .order('current_state')
   }
 
-  def user_has_badge?
-    Credly::Badge.by_programme_badge_template_ids(user.id, programme.badges.pluck(:credly_badge_template_id))
-  end
-
-  def first_stem_achievement?
-    user.achievements.in_state(:complete).with_category(Activity::FACE_TO_FACE_CATEGORY).for_programme(programme).count >= 1
-  end
-
-  def issue_badge
-    return unless programme&.badgeable? && programme&.user_enrolled?(user)
-    return unless programme.badges.active.first
-    return if user_has_badge?
-
-    Credly::IssueBadgeJob.perform_later(user.id, programme.id) if first_stem_achievement?
-  end
+  scope :belongs_to_programme, ->(programme) { joins(activity: { programme_activities: :programme }).where(activities: { programme_activities: { programme: }}) }
```

Comment:
I think this name would read better as `belonging_to_programme` since it's acting on a plural object. Same for the one below.

Reply:
Yep - I was reading it as "where belongs_to_programme" in my head, but belonging_to_programme works far better stand-alone